clouddriver-aws/src/main/java/com/netflix/spinnaker/clouddriver/aws/lifecycle/InstanceTerminationLifecycleWorker.java

```diff
@@ -227,13 +228,22 @@ private boolean updateInstanceStatus(Eureka eureka, String app, String instanceI
       try {
         eureka.updateInstanceStatus(app, instanceId, DiscoveryStatus.OUT_OF_SERVICE.getValue());
         return true;
-      } catch (RetrofitError e) {
-        final String recoverableMessage =
-            "Failed marking app out of service (status: {}, app: {}, instance: {}, retry: {})";
-        if (HttpStatus.NOT_FOUND.value() == e.getResponse().getStatus()) {
-          log.warn(recoverableMessage, e.getResponse().getStatus(), app, instanceId, retry);
-        } else if (e.getKind() == RetrofitError.Kind.NETWORK) {
-          log.error(recoverableMessage, e.getResponse().getStatus(), app, instanceId, retry, e);
+      } catch (SpinnakerServerException e) {
+        if (e instanceof SpinnakerHttpException
+            && ((SpinnakerHttpException) e).getResponseCode() == 404) {
```

Comment:
Let's try to minimize diffs
```suggestion
            && HttpStatus.NOT_FOUND.value() == ((SpinnakerHttpException) e).getResponseCode()) {
```